components/script/dom/bindings/str.rs

```diff
@@ -182,16 +182,20 @@ impl DOMString {
             return;
         }
 
-        let last_non_whitespace = match self.0.rfind(|ref c| !char::is_ascii_whitespace(c)) {
-            Some(idx) => idx + 1,
+        match self.0.rfind(|ref c| !char::is_ascii_whitespace(c)) {
+            Some(idx) => {
+                let last_non_whitespace = self.0.char_indices().find(|pair| pair.0 > idx);
+                if last_non_whitespace.is_some() {
+                    self.0.truncate(last_non_whitespace.unwrap().0);
```

Comment:
I’d recommend `if let Some(x) = y` instead of `is_some` + `unwrap`, but with the change suggested above this branch is gone.